### PR TITLE
SDIT-1315: ✨ Start to add in call out to other services

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/HmppsTemplateKotlinExceptionHandler.kt [hunk 1]
```diff
@@ -62,20 +63,3 @@ class HmppsTemplateKotlinExceptionHandler {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 }
-
-data class ErrorResponse(
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/HmppsTemplateKotlinExceptionHandler.kt:66]: use `ErrorResponse` from kotlin lib instead

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/wiremock/HmppsAuthMockServer.kt [hunk 2]
```diff
@@ -20,12 +20,10 @@ class HmppsAuthApiExtension : BeforeAllCallback, AfterAllCallback, BeforeEachCal
 
   override fun beforeAll(context: ExtensionContext) {
     hmppsAuth.start()
-    hmppsAuth.stubGrantToken()
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/wiremock/HmppsAuthMockServer.kt:23]: don't think this should be part of a `beforeAll`, should be included when needed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/WebClientConfiguration.kt [hunk 3]
```diff
@@ -3,15 +3,28 @@ package uk.gov.justice.digital.hmpps.hmppstemplatepackagename.config
...
   @Bean
   fun hmppsAuthHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(hmppsAuthBaseUri, healthTimeout)
+
+  // This is an example health bean for checking other services and should be removed / replaced
+  @Bean
+  fun templateKotlinApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(templateKotlinApiBaseUri, healthTimeout)
+
+  // This is an example bean for calling other services and should be removed / replaced
+  @Bean
+  fun templateKotlinApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager, builder: WebClient.Builder): WebClient =
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/WebClientConfiguration.kt:28]: this isn't used yet, will build on it in subsequent PRs.

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**